src/mds/Locker.cc

```diff
@@ -3634,9 +3650,19 @@ void Locker::_do_snap_update(CInode *in, snapid_t snap, int dirty, snapid_t foll
   if (xattrs) {
     dout(7) << " xattrs v" << i->xattr_version << " -> " << m->head.xattr_version
 	    << " len " << m->xattrbl.length() << dendl;
-    i->xattr_version = m->head.xattr_version;
-    auto p = m->xattrbl.cbegin();
-    decode(*px, p);
+    __u32 total = 0;
+    for (const auto &p : *px)
+      total += (p.first.length() + p.second.length());
+    if ((total + m->xattrbl.length()) >= g_conf()->mds_max_xattr_pairs_size) {
+      dout(10) << "xattr kv pairs size too big. current size: " << total
```

Comment:
This should be a warning, so please set the log level to `0` or `-1`.

Reply:
Ok.